### PR TITLE
fixes #10495 - create custom products with metadata_expire set to 1

### PR DESCRIPTION
**Pull request by jlsherrill**
also provide upgrade rake task
Red Hat products will have it set to 1 via candlepin

### CODE REVIEW

**Comment by sthirugn** [comment]
Thank you for the quick fix.

**Comment by ehelms** [comment]
What are the consequences if this is not set? The bug didn't really point me in the right direction.

**Comment by sthirugn** [comment]
Without this fix, if you perform incremental updates (on a custom repo) on a content host, the content host will fail saying something like "No packages are available to install"

**Comment by ehelms** [comment]
@jlsherrill will this not be an issue for 2.2 ?

**Comment by jlsherrill** [comment]
@ehelms yeah it is an issues for 2.2, i marked the issue to 2.2.1

File: lib/katello/tasks/upgrades/2.3/update_metadata_expire.rake [hunk 1]
```diff
@@ -0,0 +1,18 @@
+namespace :katello do
+  namespace :upgrades do
+    namespace '2.3' do
```
Comment (ehelms) [inline lib/katello/tasks/upgrades/2.3/update_metadata_expire.rake]: So is this migration accurate then if it needs to go out in 2.2.1 ?

**Comment by jlsherrill** [comment]
@ehelms updated

File: lib/katello/tasks/upgrades/2.2/update_metadata_expire.rake [hunk 2]
```diff
@@ -0,0 +1,18 @@
+namespace :katello do
+  namespace :upgrades do
+    namespace '2.3' do
```
Comment (ehelms) [inline lib/katello/tasks/upgrades/2.2/update_metadata_expire.rake]: Needs updating.

File: lib/katello/engine.rb [hunk 3]
```diff
@@ -172,6 +172,7 @@ class Engine < ::Rails::Engine
 
       load "#{Katello::Engine.root}/lib/katello/tasks/upgrades/2.1/import_errata.rake"
       load "#{Katello::Engine.root}/lib/katello/tasks/upgrades/2.2/update_gpg_key_urls.rake"
+      load "#{Katello::Engine.root}/lib/katello/tasks/upgrades/2.3/update_metadata_expire.rake"
```
Comment (ehelms) [inline lib/katello/engine.rb]: Needs updating to 2.2

**Comment by ehelms** [comment]
I assume there will also be an installer PR to add this to the upgrade steps?

**Comment by jlsherrill** [comment]
@ehelms opened: https://github.com/Katello/katello-installer/pull/224

**Comment by ehelms** [comment]
ACK

**Pull request merged**